src/libraries/Microsoft.PowerFx.Connectors/Execution/HttpFunctionInvoker.cs

```diff
@@ -63,50 +59,97 @@ public HttpRequestMessage BuildRequest(FormulaValue[] args)
+                    {
+                        switch (param.In.Value)
+                        {
+                            case ParameterLocation.Path:
+                                path = path.Replace("{" + param.Name + "}", HttpUtility.UrlEncode(valueStr));
+                                break;
+
+                            case ParameterLocation.Query:
+                                query.Append((query.Length == 0) ? "?" : "&");
+                                query.Append(param.Name);
+                                query.Append('=');
+                                query.Append(HttpUtility.UrlEncode(valueStr));
+                                break;
+
+                            case ParameterLocation.Header:
+                                headers.Add(param.Name, valueStr);
+                                break;                           
+
+                            case ParameterLocation.Cookie:
+                            default:
+                                throw new NotImplementedException($"{param.In}");
+                        }
+                    }                    
                 }
             }
 
             var url = path + query.ToString();
 
             // $$$ Not handling Body yet...
-            var request = new HttpRequestMessage(_method, url);
+            var request = new HttpRequestMessage(_method, url);            
 
             foreach (var kv in headers)
             {
                 request.Headers.Add(kv.Key, kv.Value);
             }
 
+            if (!string.IsNullOrEmpty(body))
+            {                
+                // $$$ Might need System.Net.Http.FormUrlEncodedContent()
+                request.Content = new StringContent(body, Encoding.Default, _argMapper.ContentType);                
+            }
+
             return request;
         }
 
+        private string GetBody(OpenApiParameter param, Dictionary<string, FormulaValue> map)
+        {
+            var props = GetSchemaPropertyValues(param, map);
+
+            return _argMapper.ContentType.ToLowerInvariant() switch
+            {
+                OpenApiExtensions.ContentType_XWwwFormUrlEncoded => new OpenApiFormUrlEncoder(param.Schema, props).ToUrlEncodedForm(),
+                OpenApiExtensions.ContentType_ApplicationXml => props.ToXml(param.Schema.Reference.Id ?? "Xml"),
+                _ => new OpenApiJsonSerializer(param.Schema, props).ToJson() 
+            };
+        }
+
+        private Dictionary<string, object> GetSchemaPropertyValues(OpenApiParameter param, Dictionary<string, FormulaValue> map)
+        {
+            var props = new Dictionary<string, object>();
+
+            foreach (var property in param.Schema.Properties)
+            {
+                if (map.TryGetValue(property.Key, out var paramValue))
+                {
+                    props.Add(property.Key, paramValue.ToObject());
+                }
+            }
+
+            return props;
+        }
```

Comment:
We should be able to remove this method altogether by just passing the FormulaValue directly into the FV-->json serializer.

Reply:
done, except for XML where the work isn't finished yet

Reply:
I added a comment